collectors/logsmanagement.plugin/plugin_logsmanagement_systemd.c

```diff
@@ -0,0 +1,256 @@
+        chart_data->dim_facil[19] = rrddim_add(chart_data->st_facil, "19:local3", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL); 
+        chart_data->dim_facil[20] = rrddim_add(chart_data->st_facil, "20:local4", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL); 
+        chart_data->dim_facil[21] = rrddim_add(chart_data->st_facil, "21:local5", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL); 
+        chart_data->dim_facil[22] = rrddim_add(chart_data->st_facil, "22:local6", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);
+        chart_data->dim_facil[23] = rrddim_add(chart_data->st_facil, "23:local7", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL); 
+        chart_data->dim_facil[24] = rrddim_add(chart_data->st_facil, "uknown", NULL, 1, 1, RRD_ALGORITHM_INCREMENTAL);   
+    }
+
+
+}
+
+void systemd_chart_update(struct File_info *p_file_info, struct Chart_meta *chart_meta){
+    chart_data_systemd_t *chart_data = chart_meta->chart_data_systemd;
+
+    if(chart_data->tv.tv_sec != p_file_info->parser_metrics->tv.tv_sec){
+
+        time_t lag_in_sec = p_file_info->parser_metrics->tv.tv_sec - chart_data->tv.tv_sec - 1;
+
+        chart_data->tv = p_file_info->parser_metrics->tv;
+
+        struct timeval tv = {
+            .tv_sec = chart_data->tv.tv_sec - lag_in_sec,
+            .tv_usec = chart_data->tv.tv_usec
+        };
+
+        do_num_of_logs_charts_update(p_file_info, chart_data, tv, lag_in_sec);
+
+        /* Syslog priority value - update */
+        if(p_file_info->parser_config->chart_config & CHART_SYSLOG_PRIOR){
+            if(likely(chart_data->st_prior->counter_done)){
+
+                tv.tv_sec = chart_data->tv.tv_sec - lag_in_sec;
+                
+                while(tv.tv_sec < chart_data->tv.tv_sec){
+                    for(int j = 0; j < SYSLOG_PRIOR_ARR_SIZE - 1; j++){
+                        if(chart_data->dim_prior[j])
+                            rrddim_set_by_pointer(  chart_data->st_prior, 
+                                                    chart_data->dim_prior[j], 
+                                                    chart_data->num_prior[j]);
+                    }
+                    rrddim_set_by_pointer(  chart_data->st_prior, 
+                                            chart_data->dim_prior[SYSLOG_PRIOR_ARR_SIZE - 1], // "Unknown"
+                                            chart_data->num_prior[SYSLOG_PRIOR_ARR_SIZE - 1]);
+                    rrdset_timed_done(  chart_data->st_prior, tv, true);
+                    tv.tv_sec++;
+                }
+            }
+
+            for(int j = 0; j < SYSLOG_PRIOR_ARR_SIZE; j++){
+                chart_data->num_prior[j] += p_file_info->parser_metrics->systemd->prior[j];
+                p_file_info->parser_metrics->systemd->prior[j] = 0;
+
+                if(unlikely(!chart_data->dim_prior[j] && chart_data->num_prior[j])){
+                    char dim_prior_name[4];
+                    snprintfz(dim_prior_name, 4, "%d", j);
```

Comment:
In normal situation in our codes we give as length the `SIZE -1 `, here we have the same size. 
I understand this is a minor issue, but I suggest we keep only one pattern.

Reply:
I've seen it in some parts of the code but I don't understand the reason behind this. Could you please explain to me why it is required?